extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQControllerTask.java

```diff
@@ -333,4 +336,15 @@ public static boolean writeResultsToDurableStorage(final MSQSpec querySpec)
   {
     return querySpec.getDestination() instanceof DurableStorageMSQDestination;
   }
+
+  @Override
+  public LookupLoadingSpec getLookupLoadingSpec()
+  {
+    if (getQuerySpec().getQuery().getContext().containsKey(PlannerContext.CTX_LOOKUPS_TO_LOAD)) {
+      List<String> lookupsToLoad = (List<String>) getQuerySpec().getQuery().getContext().get(PlannerContext.CTX_LOOKUPS_TO_LOAD);
+      return LookupLoadingSpec.loadOnly(new HashSet<>(lookupsToLoad));
+    } else {
+      return LookupLoadingSpec.NONE;
```

Comment:
The default behaviour being `NONE` might cause issues during a rolling upgrade since Broker is typically upgraded after Peons and MMs.

Reply:
Makes sense!

I've updated the logic to return `ALL` by default, and have started populating an empty list by default and using that to identify the `NONE` scenario.
I tested all the scenarios mentioned in the PR description post this change, everything seems to be working as expected.

Hope this works!